File: 1-two-sum/two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer, Integer> Sums = new HashMap<>();
        for (int i = 0; i < nums.length; i++) {
            int complement = target - nums[i];
            if (Sums.containsKey(complement)) {
                return new int[]{Sums.get(complement), i};
            }
            Sums.put(nums[i], i);
        }
        return new int[]{};
            
    }
}

File: 1-two-sum/two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:

        seen = {}

        for i in range(len(nums)):
            remainder = target - nums[i]
            if remainder in seen:
                return [i, seen[remainder]]
            else:
                seen[nums[i]] = i
            
